request/src/main/java/com/vimeo/networking2/interceptors/CacheControlHeaderInterceptor.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Vimeo
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.networking2.interceptors
+
+import com.vimeo.networking2.Configuration
+import okhttp3.Interceptor
+import okhttp3.Response
+
+/**
+ * Rewrite the server's cache-control header because our server sets all `Cache-Control` headers to `no-store`. To get
+ * data from the cache, we set a max age to [Configuration.cacheMaxAgeSeconds]. This prevents OkHttp from throwing a 504
+ * exception. API no longer sends a max age in the header. OkHttp determines the cache to be invalid. This is a work
+ * around for this issue.
```

Comment:
> This prevents OkHttp from throwing a 504 exception. API no longer sends a max age in the header. OkHttp determines the cache to be invalid. This is a work around for this issue.

Optional: the phrasing here feels awkward to me, maybe can rephrase this?

Reply:
Yes, I had just copied the docs from the previous java class and didn't even realize how awkward the language was. Will fix, thanks for pointing it out.